bluetooth: mesh: Fix IVU duration counter update

omkar3141 opened the pull request:
When device is first provisioned with IV Update
flag is set to 0, it should wait for minimum of
96 hours before going into IV Update In Progress
state. Such limit does not apply, if device is
provisioned with IV Update flag is set to 1.

Signed-off-by: Omkar Kulkarni <omkar.kulkarni@nordicsemi.no>

omkar3141 committed:
bluetooth: mesh: Fix IVU duration counter update

When device is first provisioned with IV Update
flag is set to 0, it should wait for minimum of
96 hours before going into IV Update In Progress
state. Such limit does not apply, if device is
provisioned with IV Update flag is set to 1.

Signed-off-by: Omkar Kulkarni <omkar.kulkarni@nordicsemi.no>

trond-snekvik commented on tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c:
```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 #include "mesh_test.h"
+#include <mesh/net.h>
```
Should be `"mesh/net.h"`, as only the include folder is treated as a system header directory

trond-snekvik commented on subsys/bluetooth/mesh/net.c:
```diff
@@ -184,11 +184,13 @@ int bt_mesh_net_create(uint16_t idx, uint8_t flags, const uint8_t key[16],
-	 * long has actually passed since the network changed its state).
+	/* If IV Update is already in progress, set minimum required hours,
+	 * since the 96-hour minimum requirement doesn't apply in this case straight
+	 * after provisioning.
```
Good catch :+1: Would you mind adding a reference to the specification section for this here (or in the commit message)? Should be 3.11.5

LingaoM replied:
I'm confused. The definition in `Mesh spec 1.0.1 section 5 provision 5.4.2.5 distribution of provision data` seems to have another meaning?

`Note: After the processing the Provisioning Data PDU from the Provisioner, the 96 hour time limits for changing the IV Update procedure state, as defined in the IV Update procedure, do not apply.`

omkar3141 replied:
Yes, this is slightly wrong and underspecified in the published. This should get rectified in the spec eventually.

trond-snekvik reviewed (changes requested):
Nice change!

Would you mind splitting the fix and the new test cases into two different commits? The change to the existing test could stay in the fix commit

omkar3141 committed:
bluetooth: mesh: Add test to check IVU counter

Adds a test to check value of the IV Update
counter at the time of provisioning based on
the status of IV Update flag. Also ensures that
the counter is reset if the mesh stack state
is reset.

Signed-off-by: Omkar Kulkarni <omkar.kulkarni@nordicsemi.no>
